hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -170,7 +170,14 @@ void tryStartJob(Function<Long, Long> executionIdSupplier) {
             return;
         }
 
-        DAG dag = deserializeDAG();
+        DAG dag;
+        try {
+            dag = deserializeDAG();
+        } catch (RuntimeException e) {
+            logger.warning("DAG deserialization failed", e);
+            finalizeJob(e);
```

Comment:
shouldn't we call `onCompleteStepCompleted()` here?

Reply:
I don't think you'll be able to complete, because verticies won't be init

Reply:
Ok I see, then it should be safe to call `onCompleteStepCompleted` instead of creating a finalizeJob method.